requirements.txt

```diff
@@ -1,7 +1,7 @@
 # pypi requirements
 minio==7.0.1
 numpy>=1.16.5
-torch>=1.0
+torch>=1.0,<=1.11.0
```

Comment:
@jannisborn is it something we strongly need? I would avoid upper bound as much as possible as they open the gates for deps hell.

Reply:
I know and I agree. But `torch==1.12` introduced a refactor of the dataset classes that moved the `*DataPipe` stuff here: https://github.com/GT4SD/gt4sd-core/blob/main/src/gt4sd/training_pipelines/torchdrug/unpatch.py#L33
to another submodule. 

It's the only way I see to guarantee a stable release. Alternatively, we could remove the 3 affected classes from the `unpatch.py` and would accept that user's downstream code might break (i dont think we use these classes ourselves anywhere so our own CI should be unaffected)

Reply:
Or we have an if/else in the import based on the torch version

Reply:
I know it's not the best, but since it will be mostly confined to the unpatch method let's put the if/else.
We can use: `pkg_resources.get_distribution("torch").version` and then compare using:

```python
import pkg_resources
from packaging import version
if version.parse("pkg_resources.get_distribution("torch").version) > version.parse(""1.11"):
    pass
else:
   pass
```